### ISSUE TITLE
Exam mode: Rejecting complaints leads to wrong points displayed to student 

### ISSUE THREAD

**Issue by jerhard** [ls1intum/Artemis#4793]
### Describe the bug

After an exam was conducted, the complaint period was started, and a complaint was filed by a student. The complaint was rejected by an instructor within the complaint period. 
For the affected exercise, the student now sees "No graded result" for the exercise, as can be seen in the following scrennshot:
![bug-exercise](https://user-images.githubusercontent.com/5951582/157420086-2be8c053-039c-4d24-b576-f64b98f48303.png)

Also, in the table that summarizes the points for their exam exercises  (see screenshot below), the students sees 0 points for the affected exercise, even though the actual points for the exercise are not zero.
![exam_overview](https://user-images.githubusercontent.com/5951582/157420006-d4c49fdf-4a5d-4711-a7b1-a89d75b317b5.png)

On the instructor side, the points for the exercise are displayed correctly, but marked as preliminary:
![preliminary](https://user-images.githubusercontent.com/5951582/157421641-22b1cdcd-6a57-4972-80f0-aafee2c7eba0.png)

### To Reproduce

1. Conduct exam
2. Grade exams
3. Publish results and start complaint period
4. Complain as a student about one exercise assessment
5. Reject complaint as an instructor
6. As a student, view your exam overview and the exercise you complained for. 
7. Observe described behavior 

### Expected behavior

The student should see the changed grading of the exercise in both the exercise view and the exam overview.

### What browsers are you seeing the problem on?

Firefox

### Additional context

The complaint period was (is) still running when the complaint was rejected and the spurious points and grading status were displayed. The exercise in question was a programming exercise. 
It is not clear whether this issue affects the display of accepted complaints and whether it affects the display of points after the end of the complaint period.

**Comment by michael-schwarz** [comment]
Additional context: We have gotten word from an affected student that this also affects the grade as shown to the students. The new grade is calculated as if they received no points for the exercise in question.

**Comment by krusche** [comment]
Thanks for reporting! We will fix it asap!

**Comment by bassner** [comment]
I tried looking into it, but I can’t reproduce it. I tried it four times locally and on TS3.

1. Created an exam with a default bubble sort programming exercise (semi-automatic assessment mode)
2. Participated and submitted
3. Assessed the exercise, gave it mediocre points
4. Complained as a student during the student review time window
5. Rejected the complaint during the student time window

For the student, it shows that the complaint has been rejected. “Your points” remained at what I set it during assessment.

@jerhard, could you share more details about the configuration of your exam and exercise? Are there any missing steps? Ideally, we would be able to recreate the exam and exercise with the exact settings you're using so we can try again to repro that issue.

**Comment by bottbenj** [comment]
OK, i had time to look into the issue a bit:
The problem exists because we planned to do second corrections, but ran out of time for the corrections and had to skip second corrections for some exercises.

To reproduce the issue as we had you need to enable second corrections but not do the second correction round, then create a complaint and reject it.

Some more digging reveals that the issue has actually nothing to do with complaints but rather happens when there is an assessment lock for that submission and thus the latest result is still pending.

Thus the result disappears for the student when visiting the assessment page to look at the complaint and not when rejecting it.

The whole issue can be fixed by canceling the assessment lock or just doing the the second correction.


In conclusion the whole thing is likely mostly by design/user error and i am not sure what can really be done about it other than maybe just adding a skip second correction button that creates dummy second correction results to avoid taking the assessment lock when viewing complaints.

**Comment by jerhard** [comment]
If second corrections are enabled, this issue also affects programming exercises that are graded only automatically. 
In other words: If you have an exam where second corrections are enabled, and where the progamming exercises are graded automatically, and the students complain about the programming exercise, and the complaint is rejected, Artemis will show 0% to the students for that exercise. 

This behavior should be changed such that the last grading is shown, also if the complaint is rejected.

**Comment by michael-schwarz** [comment]
Could you elaborate on why this is not planned? For FPV, In the past two years we have always had to tell students to disregard the points shown on Artemis after the complaints and instead consult ONLY TUMonline.

While this is a solution, telling students to ignore Artemis output doesn't feel like it's the thing one would like to have in the long-term, and creates lots of questions from students who wonder why the lost all points...

**Comment by krusche** [comment]
There was no progress on this for quite some time. We can certainly fix that if the error still occurs.

**Comment by krusche** [comment]
@michael-schwarz can you provide a few more infos how the exams are configured in FPV so that this occurs?

We need to better understand the issue in order to properly address it

**Comment by krusche** [comment]
I especially wonder about this statement:

To reproduce the issue as we had you need to enable second corrections but not do the second correction round, then create a complaint and reject it.

Why would you enable a second correction round in an exercise, but not do it?

**Comment by ekayandan** [comment]
I have identified and fixed the issue locally. Will be updating here once the fix goes live.

**Comment by krusche** [comment]
@ekayandan great. Thanks for looking into it so quickly

**Comment by ekayandan** [comment]
Fixed [#11191](https://github.com/ls1intum/Artemis/pull/11191)
